src/swaps/oneinch/OneinchSwapProvider.js

```diff
@@ -0,0 +1,263 @@
+
+    const api = new ethers.providers.StaticJsonRpcProvider(chainToRpcProviders[chainId])
+    const erc20 = new ethers.Contract(cryptoassets[quote.from].contractAddress, ERC20.abi, api)
+    const fromAddressRaw = await this.getSwapAddress(network, walletId, quote.from, quote.toAccountId)
+    const fromAddress = chains[fromChain].formatAddress(fromAddressRaw)
+    const allowance = await erc20.allowance(fromAddress, this.routerAddress)
+    const inputAmount = ethers.BigNumber.from(BN(quote.fromAmount).toFixed())
+    if (allowance.gte(inputAmount)) {
+      return {
+        status: 'APPROVE_CONFIRMED'
+      }
+    }
+
+    const callData = await axios({
+      url: this.agent + `/${chainId}/approve/calldata`,
+      method: 'get',
+      params: { tokenAddress: cryptoassets[quote.from].contractAddress }
+    })
+
+    const account = this.getAccount(quote.fromAccountId)
+    const client = this.getClient(network, walletId, quote.from, account?.type)
+    const approveTx = await client.chain.sendTransaction(callData.data)
+
+    return {
+      status: 'WAITING_FOR_APPROVE_CONFIRMATIONS',
+      approveTx,
+      approveTxHash: approveTx.hash
+    }
+  }
+
+  async sendSwap ({ network, walletId, quote }) {
+    const toChain = cryptoassets[quote.to].chain
+    const fromChain = cryptoassets[quote.to].chain
+    const chainId = ChainNetworks[toChain][network].chainId
+    if (toChain !== fromChain || !chainToRpcProviders[chainId]) return null
+
+    const account = this.getAccount(quote.fromAccountId)
+    const client = this.getClient(network, walletId, quote.from, account?.type)
+    const fromAddressRaw = await this.getSwapAddress(network, walletId, quote.from, quote.fromAccountId)
+    const fromAddress = chains[toChain].formatAddress(fromAddressRaw)
+
+    const trade = await axios({
+      url: this.agent + `/${chainId}/swap`,
+      method: 'get',
+      params: { fromTokenAddress: assets[quote.from].contractAddress || nativeAssetAddress, toTokenAddress: assets[quote.to].contractAddress || nativeAssetAddress, amount: quote.fromAmount, fromAddress: fromAddress, slippage: 0.5 }
+    })
+    await this.sendLedgerNotification(quote, account, 'Signing required to complete the swap.')
+    const swapTx = await client.chain.sendTransaction(trade.data.tx)
+    return {
+      status: 'WAITING_FOR_SWAP_CONFIRMATIONS',
+      swapTx,
+      swapTxHash: swapTx.hash
+    }
+  }
+
+  async newSwap ({ network, walletId, quote }) {
+    const approvalRequired = isERC20(quote.from)
+    const updates = approvalRequired
+      ? await this.approveTokens({ network, walletId, quote })
+      : await this.sendSwap({ network, walletId, quote })
+
+    return {
+      id: uuidv4(),
+      fee: quote.fee,
+      slippage: 50,
+      ...updates
+    }
+  }
+
+  async estimateFees ({ network, walletId, asset, accountId, txType, quote, feePrices, max }) {
+    if (txType in OneinchSwapProvider.feeUnits) {
```

Comment:
I found that one inch also returns "estimateGas" in the response for a quote
Can we use this to determine the fee estimate?

We'll need to add a buffer, (maybe * 1.5 ) 
because the gas estimate when actually sending the transaction happens inside the CAL and it is also multiplied by 1.5

Reply:
Yes, this is a good idea! Will add it